dapp/contracts/loans/InvestorLedger.sol

```diff
@@ -1,158 +1,452 @@
-pragma solidity ^0.4.11;
+pragma solidity ^0.4.17;
 
 import "../tokens/IToken.sol";
 import "../common/Math.sol";
 
 
 library InvestorLedger {
-    uint constant PERMIL = 1000;
+    /// Events
 
-    function openAccount(
-        IToken collateralToken,
-        IToken loanToken,
-        address liege,
-        uint256 totalLoanNeeded,
-        uint16 interestPermil) internal pure returns (Ledger account)
-    {
-        account.collateralToken = collateralToken;
-        account.loanToken = loanToken;
-        account.totalLoanNeeded = totalLoanNeeded;
-        account.interestPermil = interestPermil;
+    // A state transition in the loan FSM occured.
+    event StateTransistion(
+        address indexed loan,
+        State from,
```

Comment:
The compiler certainly doesn't complain. It might be because this event doesn't actually end up in the ABI definitions :). Let me copy these over to the contract definitions, as someone on the Internet pointed out [1], because blogs-as-documentation is apparently a thing in Solidity.

Is this a Solidity bug? An ABI spec bug? Why do I get no errors or warnings whatsoever. Is that by design? Is there even a design to any of this?

[1] - https://blog.aragon.one/library-driven-development-in-solidity-2bebcaf88736#7ed4

Reply:
Oh no, they do get emitted in the ABI, I just had a typo in the name and wasn't actually logging it anywhere.

Curiously, the enum gets converted to an uint8:

```json
q3k@anathema ~/Work/getline/getline/dapp $ jq '.abi[] | select(.name == "StateTransition")' < build/contracts/InvestorLedger.json 
{
  "anonymous": false,
  "inputs": [
    {
      "indexed": true,
      "name": "loan",
      "type": "address"
    },
    {
      "indexed": false,
      "name": "from",
      "type": "uint8"
    },
    {
      "indexed": false,
      "name": "to",
      "type": "uint8"
    }
  ],
  "name": "StateTransition",
  "type": "event"
}
```

Do we leave it as such, or have the states in events be defined as uint256 and explicitly cast them?

Reply:
Think that's ok

Reply:
Add a test?